pkg/apitypes/managed_tx.go

```diff
@@ -120,26 +119,34 @@ type TxHistoryActionEntry struct {
 //   - When listing back entries, the persistence layer will automatically clean up indexes if the underlying
 //     TX they refer to is not available. For this reason the index records are written first.
 type ManagedTX struct {
-	ID                 string                             `json:"id"`
-	Created            *fftypes.FFTime                    `json:"created"`
-	Updated            *fftypes.FFTime                    `json:"updated"`
-	Status             TxStatus                           `json:"status"`
-	DeleteRequested    *fftypes.FFTime                    `json:"deleteRequested,omitempty"`
-	SequenceID         *fftypes.UUID                      `json:"sequenceId"`
-	Nonce              *fftypes.FFBigInt                  `json:"nonce"`
-	Gas                *fftypes.FFBigInt                  `json:"gas"`
-	TransactionHeaders ffcapi.TransactionHeaders          `json:"transactionHeaders"`
-	TransactionData    string                             `json:"transactionData"`
-	TransactionHash    string                             `json:"transactionHash,omitempty"`
-	GasPrice           *fftypes.JSONAny                   `json:"gasPrice"`
-	PolicyInfo         *fftypes.JSONAny                   `json:"policyInfo"`
-	FirstSubmit        *fftypes.FFTime                    `json:"firstSubmit,omitempty"`
-	LastSubmit         *fftypes.FFTime                    `json:"lastSubmit,omitempty"`
-	Receipt            *ffcapi.TransactionReceiptResponse `json:"receipt,omitempty"`
-	ErrorMessage       string                             `json:"errorMessage,omitempty"`
-	History            []*TxHistoryStateTransitionEntry   `json:"history,omitempty"`
-	HistorySummary     []*TxHistorySummaryEntry           `json:"historySummary,omitempty"`
-	Confirmations      []confirmations.BlockInfo          `json:"confirmations,omitempty"`
+	ID                      string                             `json:"id"`
+	Created                 *fftypes.FFTime                    `json:"created"`
+	Updated                 *fftypes.FFTime                    `json:"updated"`
+	Status                  TxStatus                           `json:"status"`
+	DeleteRequested         *fftypes.FFTime                    `json:"deleteRequested,omitempty"`
+	SequenceID              *fftypes.UUID                      `json:"sequenceId"`
+	Nonce                   *fftypes.FFBigInt                  `json:"nonce"`
+	Gas                     *fftypes.FFBigInt                  `json:"gas"`
+	TransactionHeaders      ffcapi.TransactionHeaders          `json:"transactionHeaders"`
+	TransactionData         string                             `json:"transactionData"`
+	TransactionHash         string                             `json:"transactionHash,omitempty"`
+	PreviousTransactionHash string                             `json:"previousTransactionHash,omitempty"` // Transaction hash rarely changes after assignment, but still possible
```

Comment:
I believe this is handled already in the `history` (which took quite a lot of discussion) - keeping an eye out for what use case wasn't satisfied by that.

Reply:
Not sure I think it's exceptional for the TX hash to change - each time a new gas price estimation is done, a new TX hash would be generated. So the previous design work focussed on summarizing potentially a very long list of different hashes.

Reply:
Aha, ok so I now understand why we have this - it's because `ManagedTX` is being used for two separate purposes:
1. As the REST API structure to be returned on the query for status of a TX, and persisted to the DB
2. As the in-memory structure passed to and from the TXHandler

There is code that relies on a TXHandler carefully handling the the `previousTransactionHash` to avoid memory leaks in the confirmation manager. e.g. cleaning up the old one, when the new once changes.

This feels like a bit of a problem to me, in two ways:
- We're polluting the REST API externals of FFTM with internal information that is not designed to be persisted
- There is a complex contract with the TXHandler with regards to how it handles these fields, which really are standard computed fields.

I think at least an experimentation of whether the `ManagedTX` structure could have a split between the bit managed by FFTM, and the bit managed by the TX Manager plugin. So that a field which FFTM maintains through utility functions (like `History`,`HistorySummary` etc.) can be separated from the fields the TX Manager is responsible for updating, and the mapping to the REST object is split out.

All that said, I think the lines were already a little grey - so maybe the smallest change is:

```suggestion
	PreviousTransactionHash string                             `json:"-"`  // used internally by FFTM to track when a previous hash no longer needs to be confirmed, and can be removed from the confirmation manager
```

Reply:
@peterbroadhurst Going to pick your brain quickly before I start thinking about how to tweak the `history` update lifecycle to cope with this. 

The reason I added this field is that the confirmation manager doesn't store the ID of the managed transaction, so it's only aware of which blockchain transaction it needs to track using a hash. We had a field in `pendingState` inside the policy loop to store the previous transaction hash so that we can remove the confirmation tracking of the old hash. Is that something `history` can offer?

To use the one inside History,